client/foundries.go

```diff
@@ -784,15 +784,17 @@ func (a *Api) DeviceGet(factory, device string) (*Device, error) {
 	return &d, nil
 }
 
-func (a *Api) DeviceList(mine bool, matchTag, byFactory, byGroup, nameIlike, uuid string, page, limit int) (*DeviceList, error) {
+func (a *Api) DeviceList(
+	mine bool, matchTag, byFactory, byGroup, nameIlike, uuid, byTarget string, page, limit int,
+) (*DeviceList, error) {
 	mineInt := 0
 	if mine {
 		mineInt = 1
 	}
 	url := a.serverUrl + "/ota/devices/?"
 	url += fmt.Sprintf(
-		"mine=%d&match_tag=%s&name_ilike=%s&factory=%s&uuid=%s&group=%s&page=%d&limit=%d",
-		mineInt, matchTag, nameIlike, byFactory, uuid, byGroup, page, limit)
+		"mine=%d&match_tag=%s&name_ilike=%s&factory=%s&uuid=%s&group=%s&target_name=%s&page=%d&limit=%d",
+		mineInt, matchTag, nameIlike, byFactory, uuid, byGroup, byTarget, page, limit)
```

Comment:
we are almost to the point of the `WithOption` type pattern the Golang people use for unwieldy functions. This is an achievement :)

Reply:
yes, we are... but someone also needs to fight the lazyness pattern before that :D